### PR TITLE
Fix TestCustomGateway integration tests on OCP

### PR DESCRIPTION
**Pull request by sridhargaddam**
The test case was not configuring the platform details in the Helm values, because of which the test was failing. This PR fixes it.

Fixes: https://github.com/istio/istio/issues/52973

### CODE REVIEW

**Comment by sridhargaddam** [comment]
The arm test is flaky and the following issue is reported to track it - https://github.com/istio/istio/issues/52965

File: tests/integration/pilot/ingress_test.go [hunk 1]
```diff
@@ -728,6 +733,9 @@ resources:
     cpu: 10m
     memory: 40Mi
 `, rev)), 0o644)
+				if t.Settings().OpenShift {
+					_ = file.AppendToFile([]byte(`platform: openshift`), d)
```
Comment (dgn) [inline tests/integration/pilot/ingress_test.go:737]: as we're writing the file in the line above - wouldn't it be more efficient to append to the string before writing?

Comment (sridhargaddam) [reply tests/integration/pilot/ingress_test.go:737]: I considered that, but I felt that using `file.AppendToFile` was cleaner and easier to read, so I chose this approach. I'm open to changing it if that's preferred.

**Review by howardjohn — COMMENTED** [review]
does this test even make sense? `manifests/charts/gateways/istio-ingress` is not a supported chart to install with helm (`manifests/charts/gateway` is)

**Comment by NancyR17** [comment]
To resolve the issue, you need to ensure that your pod’s runAsUser value is within the acceptable ranges defined by the SCCs in your OpenShift cluster. This involves reviewing, creating, or modifying SCCs and ensuring they are correctly applied to the service account used by your Helm deployment.

**Comment by sridhargaddam** [comment]
> does this test even make sense? `manifests/charts/gateways/istio-ingress` is not a supported chart to install with helm (`manifests/charts/gateway` is)

Thanks for reviewing @howardjohn. Updated the PR and removed the test case.

File: tests/integration/pilot/ingress_test.go [hunk 2]
```diff
@@ -637,79 +638,6 @@ spec:
 				})
 			})
 
-			// TODO we could add istioctl as well, but the framework adds a bunch of stuff beyond just `istioctl install`
-			// that mess with certs, multicluster, etc
-			t.NewSubTest("helm").Run(func(t framework.TestContext) {
-				gatewayNs := namespace.NewOrFail(t, namespace.Config{Prefix: "custom-gateway-helm", Inject: inject})
```
Comment (keithmattix) [inline tests/integration/pilot/ingress_test.go:643]: Why are we removing a test?

Comment (sridhargaddam) [reply tests/integration/pilot/ingress_test.go:643]: I removed it based on the review comments - https://github.com/istio/istio/pull/52974#pullrequestreview-2277956693

**Review by howardjohn — APPROVED** [review]

**Pull request merged**